ToolForm.cs

```diff
@@ -246,7 +245,7 @@ private void InitializeNumericUpDown()
 
         private void button2_Click(object sender, EventArgs e)
         {
-            if (listBox2.SelectedItems.Count > 0 && comboBox2.SelectedItem != null && comboBox4.SelectedItem != null)
+            if (listBox1.SelectedItems.Count > 0 && comboBox2.SelectedItem != null && comboBox4.SelectedItem != null)
```

Comment:
why are you checking if combobox2 is null? Seems like you are not using it inside the if statement.